### PR TITLE
ADA-55 GHC Scholarship Requirement Progress Page for Recipients 

### PR DESCRIPTION
**Pull request by cachemeifyoucan0**
Files Added:
- `\acmw-db\acmwosu-db\acmw-db-app\components\ScholarshipProgressBar.jsx`
- `\acmw-db\acmwosu-db\acmw-db-app\pages\scholarshipprogress.js`
- `\acmw-db\acmwosu-db\acmw-db-app\styles\components\ScholarshipProgressBar.module.css`

How to test:
1. `git pull` on the repo
2. `git checkout ADA-55`
3. navigate to http://localhost:3000/scholarshipprogress

Added just the front end component with dummy data for now - I think it could definitely be more modular so open to suggestions!

Also, I am bad as CSS so open to suggestions there too lol

Let me know if it is not working/showing up correctly for you, I had some weird package errors at one point
image

### CODE REVIEW

File: acmw-db-app/pages/scholarshipprogress.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    function calculateMeetingProgress(testMeetingList, requiredAmountForMeetings) {
+        let percentMeetings = (testMeetingList.length / requiredAmountForMeetings) * 100;
+        return percentMeetings;
+    }
+
+    // Creating consts to set up lists of completed items for user details
+    const testScholarshipItems = testScholarshipList.map((testScholarshipList) =>
+        <li>{testScholarshipList}</li>);
```
[the remark this answers is not in the document]
Comment (smiskus) [reply acmw-db-app/pages/scholarshipprogress.js:53]: Also got that warning, I think you can fix it by doing:
`const testScholarshipItems = testScholarshipList.map((testScholarshipList, index) =>`
`<li key={index}>{testScholarshipList}</li>`

[the remark this answers is not in the document]
Comment (cachemeifyoucan0) [reply acmw-db-app/pages/scholarshipprogress.js:53]: ah shoot good catch, thank you guys!

**Comment by cachemeifyoucan0** [comment]
> I also love the progress bar! The only thing I would change is the warning stuff and maybe the CSS thing Angela mentioned. Maybe in the future this page could also include a button for members to submit a form for their volunteer hours, but that's something that could be handled later on.

I was wondering about the user functionality of this page too, that was the question I was asking Jing about and she said that the inputs are planned to be on the admin side for now - but I agree I feel like an enhancement could be like a user inputs their hours and it has like a "pending status" until an admin reviews their input or something

File: acmw-db-app/pages/scholarshipprogress.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (cachemeifyoucan0) [reply acmw-db-app/pages/scholarshipprogress.js:53]: I feel like with all these different CSS implementations there could be like a 'CSS clean up' story where someone makes everything consistent? @jgeorge37 I didn't want this as a button pop up but I wanted it to at least look consistent with the company contacts page @SnobbyDragon

[the remark this answers is not in the document]
Comment (cachemeifyoucan0) [reply acmw-db-app/pages/scholarshipprogress.js:53]: not sure if I quite understand where the minimum margin size comes from in the link you shared lol sorry 😭

[the remark this answers is not in the document]
Comment (cachemeifyoucan0) [reply acmw-db-app/pages/scholarshipprogress.js]: let me know if you guys still see the warning for some reason it didn't show up for me 😕 @smiskus @SnobbyDragon

[the remark this answers is not in the document]
Comment (jgeorge37) [reply acmw-db-app/pages/scholarshipprogress.js]: @cachemeifyoucan0 sorry I just saw that you tagged me. Yes there actually is a css cleanup story bc I figured we would probably need to refactor some styling at some point

**Comment by cachemeifyoucan0** [comment]
> Okay omg, I think I figured out the styling issues. I don't know why I just realized this but since `popup_inner` is the inside stuff, we don't have to fix that because the size of the white card should adjust based on the inside.
> 
> ```
>   #popup_inner {
>     margin: auto;
>     border-radius: 20px;
>     background: white;
>     color: black;
>     overflow: auto;
>     max-width: 500px;
>   }
> ```
> 
> this seemed to work okay. It should just keep it at 500px in size; another number could work better. auto margins won't make the card super tiny if we resize the window bc the margins will get smaller.
> 
> also changing the main content styling can align it in the center
> 
> ```
> .main {
>   padding: 0px 10px;
>   border-top: 0px;
>   flex-direction: column;
>   justify-content: center;
>   align-items: center;
> }
> ```

wait I just used the main css from `styles/Database.module.css` is that what you want to change?

**Comment by cachemeifyoucan0** [comment]
> I'm getting several warnings (in the console in the browser upon inspection; they don't show in terminal) like this:
> `Warning: validateDOMNesting(...): <tr> cannot appear as a child of <table>. Add a <tbody>, <thead> or <tfoot> to your code to match the DOM tree generated by the browser.`
> Which come from lines 73, 78, 85, and 92 in scholarshipprogress.js. Maybe the `<tr>`s need to go in` <tbody>`?

ah thank you! I'll fix that up

**Comment by SnobbyDragon** [comment]
Umm I'll have to think on that, but I feel like restyling everything all at once in the css story sounds nice. I'm pretty sure the global should be changed tho since it seems like we want like white blocks centered on top of purple background.

**Pull request merged**